tests/phpunit/tests/test-rest-request.php

```diff
@@ -39,45 +44,148 @@ public function tear_down() {
 		unset( $GLOBALS['polylang'] );
 	}
 
-	public function test_should_define_language_when_language_is_valid() {
+	/**
+	 * @dataProvider routes_provider
+	 *
+	 * @param array $data {
+	 *    @type string $method   HTTP method.
+	 *    @type string $route    Route.
+	 *    @type string $resource Resource name.
+	 * }
+	 */
+	public function test_should_define_language_when_language_is_valid( $data ) {
 		self::$model->options['default_lang'] = 'en';
+		$this->pll_rest->init();
 
-		$_REQUEST['lang'] = 'fr';
+		$request = new WP_REST_Request( $data['method'], $data['route'] );
+		$request = $this->set_lang_param( $request, 'fr' );
```

Comment:
Would `$request->set_param( 'lang', 'fr' );` not work without needing to create our own method?

Reply:
Indeed, after some research I found `set_param()` acknowledge the request method and adds the parameter accordingly.